app/Models/Forum/Topic.php

```diff
@@ -343,7 +343,13 @@ public function pollEnd()
     public function postsCount()
     {
         if ($this->postsCount === null) {
-            $this->postsCount = $this->posts()->count();
+            $query = $this->posts();
+
+            if (priv_check('HiddenForumPostView')->can()) {
+                $query->withTrashed();
```

Comment:
without this `count()` would only see the visible posts, breaking stuff for moderators who can see hidden posts